scene/main/node.cpp

```diff
@@ -1388,17 +1389,25 @@ void Node::_set_name_nocheck(const StringName &p_name) {
 	data.name = p_name;
 }
 
-void Node::set_name(const String &p_name) {
+void Node::set_name(const StringName &p_name) {
 	ERR_FAIL_COND_MSG(data.inside_tree && !Thread::is_main_thread(), "Changing the name to nodes inside the SceneTree is only allowed from the main thread. Use `set_name.call_deferred(new_name)`.");
-	String name = p_name.validate_node_name();
-
-	ERR_FAIL_COND(name.is_empty());
+	StringName new_name;
+	{
+		String input_name_str = p_name;
+		ERR_FAIL_COND(input_name_str.is_empty());
```

Comment:
Would be nice to have an error message here, how about the same as you have in `ustring.cpp`?